### PR TITLE
Allow selecting parsers to use for tagging

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by alexjurkiewicz** [comment]
I don't really understand the errors here :\

File: main.go [hunk 1]
```diff
@@ -168,6 +170,13 @@ func tagCommand() *cli.Command {
 				Usage:       "external tag group configuration file path",
 				DefaultText: "/path/to/conf/file/ (.yml/.yaml extension)",
 			},
+			&cli.StringSliceFlag{
+				Name:        parsersArgs,
+				Aliases:     []string{"i"},
+				Usage:       "IAC types to tag",
+				Value:       cli.NewStringSlice(),
```
Comment (nimrodkor) [inline main.go:177]: @alexjurkiewicz Default value should be all the possible IACs. That should also fix your test

File: src/common/runner/runner.go [hunk 2]
```diff
@@ -57,7 +57,25 @@ func (r *Runner) Init(commands *clioptions.TagOptions) error {
...
+			r.parsers = append(r.parsers, &cfnStructure.CloudformationParser{})
+			break
+		}
+	}
+	for _, value := range commands.Parsers {
+		if value == "Serverless" {
+			r.parsers = append(r.parsers, &slsStructure.ServerlessParser{})
+			break
+		}
+	}
```
Comment (nimrodkor) [inline src/common/runner/runner.go:77]: @alexberdan Please merge these for loops into a single one, with switch cases for the different values. Avoid duplicates by cleaning the `commands.Parsers` list in advance 🙏

**Review by nimrodkor — COMMENTED** [review]
Thanks @alexjurkiewicz !

See my comments please

File: main.go (hunk 1 above)
Comment (alexjurkiewicz) [reply main.go]: Ah, I thought `DefaultText` was the default value. Should have read the docs! Fixed.

File: src/common/runner/runner.go (hunk 2 above)
Comment (alexjurkiewicz) [reply src/common/runner/runner.go]: Thank you, is my new construction better?

Comment (alexjurkiewicz) [reply src/common/runner/runner.go]: Well, it's erroring, but I don't know why 😬 Can you help?

File: src/common/clioptions/cli.go [hunk 3]
```diff
@@ -25,6 +25,7 @@ type TagOptions struct {
 	TagGroups         []string `validate:"tagGroupNames"`
 	ConfigFile        string   `validate:"config-file"`
 	SkipResourceTypes []string
+  Parsers           []string
```
Comment (nimrodkor) [inline src/common/clioptions/cli.go:28]: @alexjurkiewicz I think this should be aligned with the rows above 🤔

**Review by nimrodkor — COMMENTED** [review]
Hey @alexjurkiewicz !

I debugged it a little - found the source. We have two tests (the 2 that failed, one in integration and one in unit) that bypass the actual default value and inject into the runner a custom `commands` parameter. You can see it in `integration_test.go` line 337 and `runner_test.go` line 93. If you add the `Parsers` argument manually there, should be good :)

**Comment by alexjurkiewicz** [comment]
> I debugged it a little - found the source. We have two tests (the 2 that failed, one in integration and one in unit) that bypass the actual default value and inject into the runner a custom commands parameter. You can see it in integration_test.go line 337 and runner_test.go line 93. If you add the Parsers argument manually there, should be good :)

Thank you for the explanation! Pushed this fix, sorry for the delay.

To be honest, I don't understand why it's required. Doesn't the default value I specified in the CLI options work? 🤔

**Review by nimrodkor — APPROVED** [review]
Nice.

UTs sometimes need to mock the default behavior... It's just to keep the tests in line. Thanks @alexjurkiewicz !!

**Comment by alexjurkiewicz** [comment]
Thank you for all the help!

**Pull request merged**